deps/wrapper.cc

```diff
@@ -81,12 +82,10 @@ void initialize_coreworker_worker(
             const std::string name_of_concurrency_group_to_execute,
             bool is_reattempt,
             bool is_streaming_generator) {
-          // task_executor(ray_function, returns, args);
-          int pid = task_executor(ray_function, args);
-          std::string str = std::to_string(pid);
-          auto memory_buffer = std::make_shared<LocalMemoryBuffer>(reinterpret_cast<uint8_t *>(&str[0]), str.size(), true);
+          RayObject result = task_executor(ray_function, args);
+
           RAY_CHECK(returns->size() == 1);
-          (*returns)[0].second = std::make_shared<RayObject>(memory_buffer, nullptr, std::vector<rpc::ObjectReference>());
+          (*returns)[0].second = std::make_shared<RayObject>(std::move(result));
           return Status::OK();
```

Comment:
printing debug statements shows the segfault doesn't happen until after the julia task returns
```julia
[2023-08-21 13:34:08,779 I 4983 3209946] wrapper.cc:85: Task Executing... 
[2023-08-21 13:34:09,588 I 4983 3209946] wrapper.cc:87: Task Finished... 
[2023-08-21 13:34:09,588 I 4983 3209946] wrapper.cc:90: Appending returns... 
[2023-08-21 13:34:09,588 I 4983 3209946] wrapper.cc:92: STATUS::OK 

[4983] signal (11.2): Segmentation fault: 11
in expression starting at none:1
unknown function (ip: 0xa9027bfda9014ff4)
Allocations: 5938043 (Pool: 5932656; Big: 5387); GC: 8
[2023-08-21 13:34:11,023 E 4978 3209873] (raylet) agent_manager.cc:135: The raylet exited immediately because the Ray agent failed. The raylet fate shares with the agent. This can happen because the Ray agent was unexpectedly killed or failed. Agent can fail when
- The version of `grpcio` doesn't follow Ray's requirement. Agent can segfault with the incorrect `grpcio` version. Check the grpcio version `pip freeze | grep grpcio`.
- The agent failed to start because of unexpected error or port conflict. Read the log `cat /tmp/ray/session_latest/logs/dashboard_agent.log`. You can find the log file structure here https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure.
- The agent is killed by the OS (e.g., out of memory).
```